### ISSUE TITLE
Issue Data Integrity - Product Url Key

### ISSUE THREAD

**Issue by teodorpav** [baldwin-agency/magento2-module-url-data-integrity-checker#12]
Hi,
in M 2.3.5-p2
Catalog > Data Integrity Checker > Product Url Key

Call to undefined method Baldwin\UrlDataIntegrityChecker\Model\ResourceModel\Catalog\Product\UrlKeyCollection::getMainTable()

Data Integrity - Product Url Path

Call to undefined method Baldwin\UrlDataIntegrityChecker\Model\ResourceModel\Catalog\Product\UrlPathCollection::getMainTable()

Thank you

**Comment by hostep** [comment]
Thanks for the report @teodorpav 

We've had this report before: https://github.com/baldwin-agency/magento2-module-url-data-integrity-checker/issues/8

Can you try to give me a little bit more information how this error is triggered?
Can you try to give me a stacktrace for example?
Do you have a custom module which manipulates admin grids which might cause this problem? Does disabling that module fixes this problem?

Thanks!

**Comment by hostep** [comment]
Closing due to lack of feedback.

@teodorpav: if you have more info, feel free to comment, that would be very much appreciated!

**Comment by JoergLindner** [comment]
Similar issue
An unexpected exception occured: 'We are already refreshing the product url key's, just have a little patience 🙂'
#0 /var/www/vhosts/xxxx/httpdocs/vendor/baldwin/magento2-module-url-data-integrity-checker/Console/Command/CheckProductUrlKeys.php(69): Baldwin\UrlDataIntegrityChecker\Updater\Catalog\Product\UrlKey->refresh()
#1 /var/www/vhosts/xxxx/httpdocs/vendor/symfony/console/Command/Command.php(255): Baldwin\UrlDataIntegrityChecker\Console\Command\CheckProductUrlKeys->execute()
#2 /var/www/vhosts/xxxxx/httpdocs/vendor/magento/framework/Interception/Interceptor.php(58): Symfony\Component\Console\Command\Command->run()
#3 /var/www/vhosts/xxxx/httpdocs/vendor/magento/framework/Interception/Interceptor.php(138): Baldwin\UrlDataIntegrityChecker\Console\Command\CheckProductUrlKeys\Interceptor->___callParent()
#4 /var/www/vhosts/xxxx/httpdocs/vendor/magento/framework/Interception/Interceptor.php(153): Baldwin\UrlDataIntegrityChecker\Console\Command\CheckProductUrlKeys\Interceptor->Magento\Framework\Interception\{closure}()
#5 /var/www/vhosts/xxxx/httpdocs/generated/code/Baldwin/UrlDataIntegrityChecker/Console/Command/CheckProductUrlKeys/Interceptor.php(23): Baldwin\UrlDataIntegrityChecker\Console\Command\CheckProductUrlKeys\Interceptor->___callPlugins()
#6 /var/www/vhosts/xxxxx/httpdocs/vendor/symfony/console/Application.php(1009): Baldwin\UrlDataIntegrityChecker\Console\Command\CheckProductUrlKeys\Interceptor->run()
#7 /var/www/vhosts/xxxxx/httpdocs/vendor/symfony/console/Application.php(273): Symfony\Component\Console\Application->doRunCommand()
#8 /var/www/vhosts/xxxxx/httpdocs/vendor/magento/framework/Console/Cli.php(115): Symfony\Component\Console\Application->doRun()
#9 /var/www/vhosts/xxxx/httpdocs/vendor/symfony/console/Application.php(149): Magento\Framework\Console\Cli->doRun()
#10 /var/www/vhosts/xxxxx/httpdocs/bin/magento(23): Symfony\Component\Console\Application->run()
#11 {main}

By mistake, we made a ctrl+c in cli whle running the check.
Than this happens. Seems that a task is stuck

**Comment by hostep** [comment]
Hi @JoergLindner, you can override that check by providing the `-f` flag, sorry this probably isn't documented properly but you can see it with providing the `--help` flag:

```
➜ bin/magento catalog:product:integrity:urlkey --help
Description:
  Checks data integrity of the values of the url_key product attribute.

Usage:
  catalog:product:integrity:urlkey [options]

Options:
  -f, --force           Force the command to run, even if it is already marked as already running       <================ here it is
  -h, --help            Display this help message
  -q, --quiet           Do not output any message
  -V, --version         Display this application version
      --ansi            Force ANSI output
      --no-ansi         Disable ANSI output
  -n, --no-interaction  Do not ask any interactive question
  -v|vv|vvv, --verbose  Increase the verbosity of messages: 1 for normal output, 2 for more verbose output and 3 for debug
```
